src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```
@@ -0,0 +1,199 @@
+            string http = null;     // local variable to hold most specific value
+            string https = null;
+
+
+            // get environmental variables. protocol specific take precedence over
+            // general all_*, lover case variable has precedence over upper case.
+            // note that curl uses HTTPS_PROXY but not HTTP_PROXY.
+            // For http, only http_proxy and generic variables are used.
+            string value = Environment.GetEnvironmentVariable("ALL_PROXY");
+            if (value != null)
+            {
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("all_proxy");
+            if (value != null)
+            {
+                http = value;
+                https = value;
+
+            }
+            value = Environment.GetEnvironmentVariable("http_proxy");
+            if (value != null)
+            {
+                http = value;
+            }
+            value = Environment.GetEnvironmentVariable("HTTPS_PROXY");
+            if (value != null)
+            {
+                https = value;
+            }
+
+            value = Environment.GetEnvironmentVariable("https_proxy");
+            if (value != null)
+            {
+                https = value;
+            }
+            // fail to instantiate if nothing is set.
+            // caller may pick some other proxy type.
+            if (String.IsNullOrWhiteSpace(http) && String.IsNullOrWhiteSpace(https))
+            {
+                return null;
+            }
+
+            return new HttpEnvironmentProxy(http, https);
+        }
+
+        private HttpEnvironmentProxy(string http, string https)
+        {
+
+            if (!String.IsNullOrWhiteSpace(http))
+            {
+                _http = GetUriFromString(http);
+                if (!string.IsNullOrWhiteSpace(_http.UserInfo))
+                {
+                    String[] s = _http.UserInfo.Split(':', 2);
```

What if `UserInfo` doesn't have a `:`?

---

I'm not sure what is best behavior. I guess it show throw exception and I think it will right now. 
I guess we can customize it to be more specific. Other option would be to silently ignore it. 
I'll do more testing with curl.